packages/hooks/src/useEventListener.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useRef } from 'react';
+
+interface UseEventOptions {
+  options?: object;
+  dependencies?: Array<any>;
+  enabled?: boolean;
+  element?: Document | HTMLElement;
+}
+
+/**
+ * Hook to subscribe to an event listener.
+ * @param type Represents the event type to listen for.
+ * @param eventCallback Event listener callback function.
+ * @param optional Optional third argument passed to function with implementation specifications
+ * @param optional.options Parameter to specify options passed to the eventListener.
+ * @param optional.dependencies Array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
+ * @param optional.enabled Determines whether or not the useEffect hook should run.
+ * @param optional.element Value to be passed as target of event handler, will default to document.
+ */
+export default function useEventListener(
+  type: string,
+  eventCallback: (e) => void,
+  {
+    options = undefined,
+    dependencies = [type, eventCallback],
```

Comment:
Default actually can just be `[enabled, type]` since we're using that `memoizedEventCallback` hack to not worry about `eventCallback` changing.

Reply:
👍